org.jacoco.core/src/org/jacoco/core/JaCoCo.java

```diff
@@ -30,14 +30,16 @@
 	public static final String RUNTIMEPACKAGE;
 
 	/** ASM API version */
-	public static final int ASM_API_VERSION = Opcodes.ASM5;
```

Comment:
This should only get compiled into JaCoCo classes and not be used externally. For internal classes inlining is perfectly ok because the compiled version of JaCoCo has specifically been tested for this ASM version.

Instead of changing the initialization scheme here I would rather remove it from the API and move it to some internal classes (e.g. InstrSupport).